### PR TITLE
Update cyvcf2 to 0.31.1

### PR DESCRIPTION
**Pull request by BiocondaBot**
Update [`cyvcf2`](https://bioconda.github.io/recipes/cyvcf2/README.html): **0.31.0** &rarr; **0.31.1**

[install with bioconda](http://bioconda.github.io/recipes/cyvcf2/README.html) [Conda](https://anaconda.org/bioconda/cyvcf2/files)

Info | Link or Description
-----|--------------------
Recipe | [`recipes/cyvcf2`](https://github.com//bioconda/bioconda-recipes/tree/bump/cyvcf2/recipes/cyvcf2) (click to view/edit other files)
Summary | A cython wrapper around htslib built for fast parsing of Variant Call Format (VCF) files
Home | [https://github.com/brentp/cyvcf2/](https://github.com/brentp/cyvcf2/)
Releases |[https://pypi.org/pypi/cyvcf2/json](https://pypi.org/pypi/cyvcf2/json)

***

This pull request was automatically generated (see [docs](https://bioconda.github.io/contributor/updating.html)).

### CODE REVIEW

**Comment by aryarm** [comment]
As of [this change](https://github.com/brentp/cyvcf2/compare/v0.31.0...v0.31.1#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711), numpy >=2.0 is required to build `cyvcf2` on python 3.12, but the CI seems to be using numpy 1.26.4 [according to this](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=61858&view=logs&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=34540139-28f3-5244-2e84-5ff07800d4d9&l=115)

**Comment by aryarm** [comment]
It's strange: all of the tests work except for py3.8 on OSX...

even though [there is a built wheel for py3.8](https://github.com/brentp/cyvcf2/actions/runs/9637450938/job/26576698247) already uploaded to pypi

File: recipes/cyvcf2/meta.yaml [hunk 1]
```diff
@@ -35,7 +36,8 @@ requirements:
     - zlib
     - htslib >=1.10
     - openssl
-    - numpy
+    - numpy >=2.0.0  # [py >= 39]
+    - numpy >=1.16.0,<2.0.0  # [py < 39]
```
Comment (aryarm) [inline recipes/cyvcf2/meta.yaml:40]:
```suggestion
    - numpy  # [py < 39]
```
Any compatible version of numpy is fine and numpy 2.0 cannot be installed with python 3.8 anyway

File: recipes/cyvcf2/meta.yaml [hunk 2]
```diff
@@ -14,7 +14,8 @@ source:
     - patches/setup.py.patch
 
 build:
-  number: 1
+  number: 0
+  skip: True  # [py < 37]
   script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --no-cache-dir -vvv
```
Comment (aryarm) [inline recipes/cyvcf2/meta.yaml:19]:
```suggestion
  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --no-cache-dir -vvv
  script_env:
    - CFLAGS=-Wno-implicit-function-declaration -Wno-int-conversion  # [osx and py == 38]
```

This silences some errors that appear on osx with python 3.8 (apparently due to XCode being strict):
https://github.com/scikit-image/scikit-image/issues/5051#issuecomment-729795085

These same errors appear as warnings in the Linux builds with python 3.8

**Review by aryarm — COMMENTED** [review]
I spent some time playing around with the recipe in https://github.com/aryarm/aryarm-conda/pull/8 and found that the below combination of changes might fix the failing build 🤞

**Comment by aryarm** [comment]
@BiocondaBot update

**Comment by mencian** [comment]
Seems that did the trick, thanks for looking into this @aryarm!

**Comment by aryarm** [comment]
Great! Thanks for trying out my suggestions, @mencian 🙌

Out of curiosity, do you know anything about those `died with <Signals.SIGABRT: 6>` errors we were getting? I'd like to become a bioconda maintainer one day, so I'm just trying to understand how you knew what to do

**Comment by mencian** [comment]
`died with <Signals.SIGABRT: 6>` is _usually_ a temporary error that only seems to happen with the macOS runner. I can't give a definite answer as to why it happens but updating the master branch (in this case it took 3 tries) to re-rerun the macOS build almost always solves the issue.

**Comment by aryarm** [comment]
ah ok that makes sense, thanks!

**Pull request merged**